File: Project2数组乘法对比/src/gptcode.java
import java.util.Random;

public class gptcode {
    public static void main(String[] args) {
        final int N = 100000000;
        double[] a = new double[N];
        double[] b = new double[N];
        Random random = new Random();
        for (int i = 0; i < N; ++i) {
            a[i] = random.nextDouble();
            b[i] = random.nextDouble();
        }

        long start = System.currentTimeMillis();
        double sum = 0.0;
        for (int i = 0; i < N; ++i) {
            sum += a[i] * b[i];
        }
        long end = System.currentTimeMillis();
        long duration = end - start;

        System.out.println("Java dot product computation took " + duration + " milliseconds");
        System.out.println("Dot product result: " + sum);
    }
}

File: Project2数组乘法对比/c_code/gptcode.cpp
#include <iostream>
#include <chrono>
#include <vector>
#include <random>

using namespace std;
using namespace chrono;

int main()
{
    const int N = 1000000000;
    vector<double> a(N), b(N);
    default_random_engine generator;
    uniform_real_distribution<double> distribution(0.0, 1.0);
    for (int i = 0; i < N; ++i) {
        a[i] = distribution(generator);
        b[i] = distribution(generator);
    }

    auto start = high_resolution_clock::now();
    double sum = 0.0;
    for (int i = 0; i < N; ++i) {
        sum += a[i] * b[i];
    }
    auto end = high_resolution_clock::now();
    auto duration = duration_cast<milliseconds>(end - start);

    cout << "C++ dot product computation took " << duration.count() << " milliseconds" << endl;
    cout << "Dot product result: " << sum << endl;

    return 0;
}